var/spack/repos/builtin/packages/vtk-m/package.py

```diff
@@ -231,17 +231,22 @@ def cmake_args(self):
             if "+cuda_native" in spec:
                 options.append("-DVTKm_ENABLE_CUDA:BOOL=ON")
                 options.append("-DCMAKE_CUDA_HOST_COMPILER={0}".format(env["SPACK_CXX"]))
-                if "cuda_arch" in spec.variants:
-                    cuda_value = spec.variants["cuda_arch"].value
-                    cuda_arch = cuda_value[0]
-                    if cuda_arch in gpu_name_table:
-                        vtkm_cuda_arch = gpu_name_table[cuda_arch]
-                        options.append("-DVTKm_CUDA_Architecture={0}".format(vtkm_cuda_arch))
+
+                cuda_archs = spec.variants["cuda_arch"].value
+                if spec.satisfies("@1.9.0: ^cmake@3.18:"):
+                    str_cuda_archs = ";".join(str(e) for e in cuda_archs)
+                    options.append(f"-DCMAKE_CUDA_ARCHITECTURES={str_cuda_archs}")
```

Comment:
Is this something worth pushing to the `CMakeBuilder` base class as a method?

Reply:
Probably yes, should I add that change in this PR?

Reply:
Here we should change the call to account for the new method in `CMakeBuilder`